File: src/main/java/seremis/geninfusion/soul/entity/render/ModelEntitySoulCustom.java
package seremis.geninfusion.soul.entity.render;

import net.minecraft.client.model.ModelBase;
import net.minecraft.entity.Entity;
import seremis.geninfusion.api.soul.IEntitySoulCustom;
import seremis.geninfusion.soul.TraitHandler;

public class ModelEntitySoulCustom extends ModelBase {

    @Override
    public void render(Entity entity, float timeModifier, float limbSwing, float specialRotation, float rotationYawHead, float rotationPitch, float scale) {
        TraitHandler.render((IEntitySoulCustom) entity, timeModifier, limbSwing, specialRotation, rotationYawHead, rotationPitch, scale);
    }
}

File: src/main/scala/seremis/geninfusion/soul/entity/render/ModelEntitySoulCustom.scala
package seremis.geninfusion.soul.entity.render

import net.minecraft.client.model.ModelBase
import net.minecraft.entity.Entity
import seremis.geninfusion.api.soul.IEntitySoulCustom
import seremis.geninfusion.soul.TraitHandler

class ModelEntitySoulCustom extends ModelBase {

    override def render(entity: Entity, timeModifier: Float, limbSwing: Float, specialRotation: Float, rotationYawHead: Float, rotationPitch: Float, scale: Float) {
        TraitHandler.render(entity.asInstanceOf[IEntitySoulCustom], timeModifier, limbSwing, specialRotation, rotationYawHead, rotationPitch, scale)
    }
}